File: solutions/Question 3512/Solution4.py
# Problem: 3512. Minimum Operations to Make Array Sum Divisible by K
# Difficulty: Easy
# Topics: Array, Math
# URL: https://leetcode.com/problems/minimum-operations-to-make-array-sum-divisible-by-k/
# Runtime: 0 ms
# Memory: 18.1 MB
# Solved via CodePath Auto-Committer

class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        x=sum(nums)
        if x%k==0:
            return 0
        else:
            if ((x%k)+x)%k==0:
                return k-(x%k)
            else:
                return x%k
        

File: solutions/Question 1/Solution4.cpp
/*
 * Problem: 1. Two Sum
 * Difficulty: Easy
 * Topics: Array, Hash Table
 * URL: https://leetcode.com/problems/two-sum/
 * Runtime: 71 ms
 * Memory: 14.1 MB
 * Solved via CodePath Auto-Committer
 */

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int x = 0;
        vector<int> ans;
        for(int i = 0;i<nums.size();i++){
            for(int j = i+1 ; j<nums.size();j++){
                if(nums[i]+nums[j] == target){
                    ans.push_back(i);
                    ans.push_back(j);
                    x++;
                    break;
                }
            }
            if(x) break;
        }
        return ans;
    }
};